src/js/lib/utils.js

```diff
@@ -476,6 +471,19 @@ var utils = {
             seconds = ("0" + date.getSeconds()).slice(-2);
 
         return year + '-' + month + '-' + day + '_' + hour + '.' + minutes + '.' + seconds;
+    },
+
+    getPathsRecursevly: function(dir, paths = []) {
+        if (fs.existsSync(dir)) {
+            fs.readdirSync(dir).forEach(function (file) {
+                const filepath = path.join(dir, file);
+                if (fs.statSync(filepath).isDirectory()) {
+                    return utils.getPathsRecursevly(filepath, paths);
+                }
+                paths.push(filepath);
+            });
+        }
+        return paths;
```

Comment:
This approach works, but I recommend against using recursion to solve problems like walking a file tree.  Generally, an iterative solution is faster and consumes less memory than a recursive one.  

I recommend you re-implement this as an iterative solution, like the following:

```javascript
getPaths: function(root) {
    let dirsToCheck = [root];
    let paths = [];
    while (dirsToCheck.length > 0) {
        const dir = dirsToCheck.pop();
        fs.readdirSync(dir).forEach(function (file) {
            const filepath = path.join(dir, file);
            if (fs.statSync(filepath).isDirectory()) {
                dirsToCheck.push(filepath);
            } else {
                paths.push(filepath);
            }
        });
    }
    return paths;
}
```

Reply:
Thanks for the advice and the code :)